### PR TITLE
Bump `react-select` and Update `SearchableDropdown`

### PR DESCRIPTION
**Pull request by jcq**
Connects #14490

### Description
This bumps `react-select` to the current version (3.1.0), updates `SearchableDropdown` to match new syntax and imports, sets up new SCSS styles due to changes in `react-select`, and adjusts the async functions of the few components that use it when calling `SearchableDropdown`. 

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Styling of controls remain visually the same
- [ ] Components that use `async` prop of `SearchableDropdown` are updated to return just the array of options rather than wrapped in an object
- [ ] Accessibility checks pass

### CODE REVIEW

File: client/test/app/hearings/components/modalForms/HearingTime.test.js [hunk 1]
```diff
@@ -30,45 +29,35 @@ describe('HearingTime', () => {
...
-      form.find('input').exists({ checked: true, value: "other" })
-    ).toBe(true);
+    expect(form.find('input').exists({ checked: true, value: 'other' })).toBe(true);
 
     // Expect dropdown to be populated with correct time
     expect(form.exists('SearchableDropdown')).toBe(true);
-    expect(
-      form.find('Select').exists({ value: '13:45' })
-    ).toBe(true);
+    console.log(form.debug());
```
Comment (ferristseng) [inline client/test/app/hearings/components/modalForms/HearingTime.test.js:55]: `console.log` here

File: client/test/app/hearings/components/modalForms/HearingTime.test.js [hunk 2]
```diff
@@ -30,45 +29,35 @@ describe('HearingTime', () => {
...
   });
 
   test('Matches snapshot when other time is selected', () => {
-    const form = mount(
-      <HearingTime
-        value={'13:45'}
-      />
-    );
+    const selectedTime = '13:45';
+    const option = HEARING_TIME_OPTIONS.find(({ value }) => value === selectedTime);
```
Comment (ferristseng) [inline client/test/app/hearings/components/modalForms/HearingTime.test.js:44]: Is this handled by the compatibility code from lines 123-129 in `SearchableDropdown`?

**Comment by ferristseng** [comment]
Oh, I did notice some weird UI issues. 

@sahalliburton: You might want to take a look at this too because the line in the hearing timezone dropdown isn't showing for me.

## 1

Screen Shot 2020-07-01 at 10 58 20 AM

## 2

Screen Shot 2020-07-01 at 11 06 32 AM

**Comment by ferristseng** [comment]
One more thing that I realized is my `client/test/app/hearings/components/VirtualHearings/__snapshots__/Timezone.test.js.snap` file looks different 🤔

File: client/app/components/SearchableDropdown.jsx [hunk 3]
```diff
@@ -39,7 +47,7 @@ class SearchableDropdown extends React.Component {
     }
     // don't set value in state if creatable is true
     if (!this.props.selfManageValueState) {
-      this.setState({ value: newValue });
+      this.setState({ value: this.props.clearOnSelect ? null : newValue });
```
[the remark this answers is not in the document]
Comment (ferristseng) [reply client/app/components/SearchableDropdown.jsx:50]: Nevermind, I think I got this. I tested out the `OrganizationUsers` dropdown, and it makes sense now!

File: client/test/app/hearings/components/modalForms/HearingTime.test.js (hunk 2 above)
Comment (jcq) [reply client/test/app/hearings/components/modalForms/HearingTime.test.js:44]: Yup, and I went ahead and updated propTypes for `SearchableDropdown` to show that we do still support string value (though we should consider it deprecated)

**Pull request merged**